rskj-core/src/test/java/co/rsk/peg/lockingcap/LockingCapStorageProviderImplTest.java

```diff
@@ -0,0 +1,111 @@
+    }
+
+    @Test
+    void getLockingCap_whenAPreviousLockingCapValueExistsInStorage_shouldReturnPreviousValue() {
+        // Arrange
+        Coin previousLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(previousLockingCap);
+        lockingCapStorageProvider.save(activations);
+        // Recreate LockingCapStorageProvider to load the previous value from storage
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+
+        // Assert
+        assertEquals(Optional.of(previousLockingCap), actualLockingCap);
+    }
+
+    @Test
+    void getLockingCap_prePapyrus200_whenLockingCapIsNotSet_shouldReturnEmpty() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+
+        // Assert
+        assertEquals(Optional.empty(), actualLockingCap);
+    }
+
+    @Test
+    void setLockingCap_whenIsSetANewLockingCapValue_shouldSetLockingCap() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+
+        // Act
+        lockingCapStorageProvider.setLockingCap(newLockingCap);
+
+        // Assert
+        assertEquals(Optional.of(newLockingCap), lockingCapStorageProvider.getLockingCap(activations));
+    }
+
+    @Test
+    void save_whenIsSavedANewLockingCapValue_shouldSaveLockingCap() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(newLockingCap);
+
+        // Act
+        lockingCapStorageProvider.save(activations);
+
+        // Assert
+        // Recreate LockingCapStorageProvider to load the previous value from storage and make sure it was saved
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        assertEquals(Optional.of(newLockingCap), lockingCapStorageProvider.getLockingCap(activations));
+    }
+
+    @Test
+    void save_prePapyrus200_whenIsAttemptedSaveANewLockingCapValue_shouldNotSaveLockingCap() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(newLockingCap);
+
+        // Act
+        lockingCapStorageProvider.save(activations);
+
+        // Assert
+        // Recreate LockingCapStorageProvider to load the previous value from storage and make sure it was not saved
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        assertEquals(Optional.empty(), lockingCapStorageProvider.getLockingCap(activations));
+    }
+}
```

Comment:
Remember always to leave an empty line at the end of the file.

Reply:
Done!